stan/math/prim/mat.hpp

```diff
@@ -46,6 +46,8 @@
 #include <stan/math/prim/mat/err/constraint_tolerance.hpp>
 #include <stan/math/prim/mat/err/validate_non_negative_index.hpp>
 
+#include <stan/math/prim/mat/vectorize/apply_scalar_unary.hpp>
```

Comment:
I believe the includes here are alphabetical after the meta traits. Is there a reason this needs to be included before the `stan/math/prim/mat/fun/` files? If not, could we move it to alphabetical order?